lib/test/KettleTestUtils.http.js

```diff
@@ -0,0 +1,185 @@
+                if (fluid.isDestroyed(cookieJar)) {
+                    fluid.fail("Stored cookie in destroyed jar");
+                }
+                cookieJar.cookie = cookie;
+                // Use connect's cookie parser with set secret to parse the
+                // cookies from the kettle.server.
+                pseudoReq = {
+                    headers: {
+                        cookie: cookie[0]
+                    }
+                };
+                // pseudoReq will get its cookies and signedCookies fields
+                // populated by the cookie parser.
+                cookieJar.parser(pseudoReq, {}, fluid.identity);
+            }
+            callback(data, that, pseudoReq);
+        });
+    });
+
+    // See http://stackoverflow.com/questions/6658761/how-to-destroy-a-node-js-http-request-connection-nicely and 
+    // https://github.com/joyent/node/blob/master/lib/_http_client.js#L136 - probably unnecessary in modern node
+    req.shouldKeepAlive = false;
+    
+    req.on("error", function (err) {
+        jqUnit.fail("Error making request to " + requestOptions.path + ": " + err.message);
+    });
+
+    if (model) {
+        req.write(model);
+    }
+
+    req.end();
+};
+
+/** Accepts a structure containing:
+ *  message {String} The assertion message
+ *  expected {Object} The expected response body as JSON
+ *  string {String} The received response body
+ *  request {Component} The http request component which has fired
+ *  statusCode {Number} The expected status code (defaults to 200 if missing)
+ */
+kettle.test.assertJSONResponse = function (options) {
+    var statusCode = options.statusCode || 200;
+    var data = kettle.JSON.parse(options.string);
+    jqUnit.assertDeepEq(options.message, options.expected, data);
+    jqUnit.assertEquals(options.message + " statusCode", statusCode, options.request.nativeResponse.statusCode);
+};
+
+/** Accepts a structure containing:
+ *  message {String} The assertion message
+ *  errorTexts {String/Array of String} Strings which are expected to appear within the text of the "message" field of the response
+ *  string {String} The received response body
+ *  request {Component} The http request component which has fired
+ *  statusCode {Number} The expected status code (defaults to 500 if missing)
+ *  plainText {Boolean} If a plain text (and not a JSON) response is expected
+ */
+kettle.test.assertErrorResponse = function (options) {
+    var statusCode = options.statusCode || 500;
+    var data = options.plainText ? options.string : kettle.JSON.parse(options.string);
+    if (!options.plainText) {
+        jqUnit.assertEquals(options.message + " isError field set", true, data.isError);
+    }
+    var errorTexts = fluid.makeArray(options.errorTexts);
+    fluid.each(errorTexts, function (errorText) {
+        jqUnit.assertTrue(options.message + " - message text \"" + data.message + "\" must contain " + errorText, data.message.indexOf(errorText) >= 0);
```

Comment:
As we chatted about Antranig, I believe that the expression "data.message.indexOf()" on this line will fail in the case that options.plainText is true. In that case the "data" variable will be assigned to "options.string" (line 175) and will not be an object with a "message" property.